doc/dev/doc/network.md

```diff
@@ -13,18 +13,37 @@ You can specify multiple addresses, with or without the port information. If the
 
-The client has two operation modes because of the distributed nature of the data and cluster: smart and unisocket. Smart routing is enabled by default, and is controlled by the `hazelcast.networking.smartRouting` configuration option, but you may want to enable unisocket mode.
+> [!NOTE]
+> Routing options are renamed and introduced new routing modes, such as `MultiMember`. Hence, `@Hazelcast.Networking.SmartRouting` API is deprecated and will be removed in the next major release. Please use the new `@Hazelcast.Networking.RoutingModes` API instead.
+
+
+### Routing Mode
 
-### Smart Routing
+Routing mode allows you to choose the routing strategy of the client. 
 
-In the smart mode, the clients connect to each cluster member. Since each data partition uses the well known and consistent hashing algorithm, each client can send an operation to the relevant cluster member, which increases the overall throughput and efficiency. Smart mode is the default mode.
+There are three routing modes:
+* `@Hazelcast.Networking.RoutingModes.SingleMember` aka Unisocket mode.
+* `@Hazelcast.Networking.RoutingModes.MultiMember` connects to subset of the cluster.
+* `@Hazelcast.Networking.RoutingModes.AllMembers` connects to all members in the cluster aka Smart mode.
 
-### Unisocket Client
+#### Single Member Routing Mode
 
 For some cases, the clients can be required to connect to a single member instead of each member in the cluster. Firewalls, security or some custom networking issues can be the reason for these cases.
 
 In the unisocket client mode, the client will only connect to one of the configured addresses. This single member will behave as a gateway to the other members. For any operation requested from the client, it will redirect the request to the relevant member and return the response back to the client returned from this member.
+#### Multi Member Routing Mode
+Multi member routing mode is a mode where the client connects to a subset of the cluster members. This mode behaves based
+on given strategy on `@Hazecast.Networking.NetworkingOptions.RoutingMode.Strategy`.
+
+Available strategies are:
+* `@Hazelcast.Networking.RoutingStrategy.PartitionGroups` which is the default strategy. Clients connects to subset of the cluster based on partition groups which contains the current connected member. 
+
+Please note that server should support this routing mode which is available with Hazelcast 5.5.0.
+
+#### All Members Routing Mode
+In all members mode, the clients connect to each cluster member. Since each data partition uses the well known and consistent hashing algorithm, each client can send an operation to the relevant cluster member, which increases the overall throughput and efficiency. Smart mode is the default mode.
```

Comment:
```suggestion
In all members mode, the clients connect to each cluster member. Since each data partition uses the well known and consistent hashing algorithm, each client can send an operation to the relevant cluster member, which increases the overall throughput and efficiency. This mode is the default mode.
```

Reply:
Fixed.